tests/io/v3/vm/tags/test_tags_api.py

```diff
@@ -0,0 +1,592 @@
+    assert res['id'] == category_id
+
+
+@responses.activate
+def test_edit_category(api):
+    '''
+    Test case for tags edit_category method
+    '''
+    name: str = 'Test Location'
+    description: str = 'Test location category for tags'
+    category_id: str = '54595f6e-648c-4570-b967-9b4e6a947634'
+    test_response = {
+        'id': category_id,
+        'created_at': '2021-12-29T17:01:05.462Z',
+        'created_by': 'jyoti.patel@crestdatasys.com',
+        'updated_at': '2021-12-29T17:44:11.706Z',
+        'updated_by': 'jyoti.patel@crestdatasys.com',
+        'name': name,
+        'description': description,
+        'reserved': False,
+        'value_count': 0
+    }
+    payload: dict = {
+        'name': name,
+        'description': description
+    }
+    responses.add(
+        responses.PUT,
+        re.compile(f'{BASE_URL}/categories/{category_id}'),
+        match=[responses.matchers.json_params_matcher(payload)],
+        json=test_response
+    )
+    res = api.v3.vm.tags.edit_category(
+        category_id=category_id,
+        name=name,
+        description=description
+    )
+    assert isinstance(res, dict)
+    assert res['id'] == category_id
+    assert res['name'] == name
+    assert res['description'] == description
+
+
+@responses.activate
+def test_delete_category(api):
+    '''
+    Test case for tags delete_category method
+    '''
+    category_id: str = '54595f6e-648c-4570-b967-9b4e6a947634'
+    responses.add(
+        responses.DELETE,
+        re.compile(f'{BASE_URL}/categories/{category_id}')
+    )
+    res = api.v3.vm.tags.delete_category(
+        category_id=category_id
+    )
+    assert res is None
+
+
+@responses.activate
+def test_details(api):
+    '''
+    Test case for tags details method
+    '''
+    value_id: str = '60bfce80-f695-4ed0-bd47-ec9e5b2946e5'
+    test_response: dict = {
+        'id': value_id,
+        'created_at': '2021-12-30T05:34:46.626Z',
+        'created_by': 'jyoti.patel@crestdatasys.com',
+        'updated_at': '2021-12-30T05:34:46.626Z',
+        'updated_by': 'jyoti.patel@crestdatasys.com',
+        'category_uuid': '4c5f6aaf-22fe-458d-a7a3-76c60e90147a',
+        'value': 'Chicago',
+        'type': 'static',
+        'category_name': 'Location',
+        'category_description': 'category for particuler location',
+        'assignment_count': 0,
+        'access_control': {
+            'current_user_permissions': [
+                'ALL',
+                'CAN_EDIT',
+                'CAN_SET_PERMISSIONS',
+                'CAN_USE'
+            ],
+            'defined_domain_permissions': [
+                'ALL',
+                'CAN_EDIT',
+                'CAN_SET_PERMISSIONS',
+                'CAN_USE'
+            ],
+            'all_users_permissions': [],
+            'current_domain_permissions': [
+                {
+                    'id': 'e9f23194-adb7-4c02-8632-615c694c787e',
+                    'name': 'jyoti.patel@crestdatasys.com',
```

Comment:
update here and all other instances with some test user `user1@gmail.com`

Reply:
done.